pkg/controllers/managementagent/monitoring/operatorHandler.go

```diff
@@ -40,32 +40,37 @@ func (h *operatorHandler) syncCluster(key string, obj *mgmtv3.Cluster) (runtime.
 		return obj, nil
 	}
 
-	var newCluster *mgmtv3.Cluster
-	var err error
+	cpy := obj.DeepCopy()
+	var err, updateErr error
 	//should deploy
 	if obj.Spec.EnableClusterAlerting || obj.Spec.EnableClusterMonitoring {
-		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(obj, func() (runtime.Object, error) {
-			cpy := obj.DeepCopy()
+		newObj, err := v32.ClusterConditionPrometheusOperatorDeployed.Do(cpy, func() (runtime.Object, error) {
```

Comment:
This `err` shadows the one created above so the if check below will never get an error from here. This can be fixed by either calling this error something else and setting the outer scope to this one or declaring `newObj` at the outer scope so the `:=` can be dropped